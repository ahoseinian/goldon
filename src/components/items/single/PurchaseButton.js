import React, {PropTypes} from 'react';
import {connect} from 'react-redux'
import styled from 'styled-components'
import {Button} from '../../common/Button'
import Icon from '../../common/Icon'
import {addItemToCart} from '../../../actions/index'
import findById from '../../../reducers/items/findById'

const AddButton = styled(Button)`
  padding: 1rem 2rem;
  width: 100%;
  height: 100%;
`

export const PurchaseButton = ({item, addItemToCart, cartItems}) => {
  const itemInCart = findById(cartItems, item.id)
  return (
    <AddButton className="btn-block" onClick={addItemToCart.bind(null, item)}>
      <Icon name="plus"/>
      <div className="d-inline mr-2">
        {!itemInCart
          ? <span className="buy-text">
              خرید
            </span>
          : <div className="d-inline">
            <span>افزودن تعداد درون سبد به</span>
            <u className="mr-2 item-quanity-plus">
              {itemInCart.quantity + 1}
            </u>
          </div>}
      </div>
    </AddButton>
  )
}

PurchaseButton.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired
  }).isRequired,
  addItemToCart: PropTypes.func.isRequired,
  cartItems: PropTypes.array.isRequired
};

export default connect((state, {item}) => ({cartItems: state.cart.items}), ({addItemToCart}))(PurchaseButton)
