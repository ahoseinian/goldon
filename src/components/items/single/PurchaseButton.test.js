import React from 'react'
import {shallow} from 'enzyme'
import {PurchaseButton} from './PurchaseButton'
import toJson from 'enzyme-to-json'

describe('PurchaseButton', () => {
  it('shows buy button when item does not exist in cart', () => {
    const wrapper = shallow(<PurchaseButton
      item={{id:'23df'}}
      addItemToCart={jest.fn()}
      cartItems={[]}
    />)

    expect(
      wrapper.find('.buy-text')
    ).toHaveLength(1)

    expect(toJson(
      wrapper
    )).toMatchSnapshot()
  })


  describe('item exists in cart', () => {
    const item = { id: 'test' }
    const cartItems = [{id: 'test', quantity: 3}, {id: 'a3f32', quantity: 2}]
    const wrapper = shallow(<PurchaseButton
      item={item}
      addItemToCart={jest.fn()}
      cartItems={cartItems}
    />)

    it('has quantity of item in cart +1', () => {
      expect(
        wrapper.find('.item-quanity-plus').first().text()
      ).toEqual('4')
    })

    it('does not render plus button when item exists in cart', () => {
      expect(
        wrapper.find('.buy-text')
      ).toHaveLength(0);
    })

  })


})
