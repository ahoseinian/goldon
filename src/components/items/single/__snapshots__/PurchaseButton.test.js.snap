exports[`PurchaseButton shows buy button when item does not exist in cart 1`] = `
<styled.button
  className="btn-block"
  onClick={[Function]}>
  <Icon
    name="plus" />
  <div
    className="d-inline mr-2">
    <span
      className="buy-text">
      خرید
    </span>
  </div>
</styled.button>
`;
